lib/charms/observability_libs/v1/kubernetes_service_patch.py

```diff
@@ -294,6 +323,16 @@ def _delete_and_create_service(self, client: Client):
         client.delete(Service, self._app, namespace=self._namespace)
         client.create(service)
 
+    def _delete_and_create_lb_service(self, client: Client):
+        try:
+            service = client.get(Service, self.service_name, namespace=self._namespace)
+            service.metadata.name = self.service_name  # type: ignore[attr-defined]
+            service.metadata.resourceVersion = service.metadata.uid = None  # type: ignore[attr-defined]   # noqa: E501
+            client.delete(Service, self.service_name, namespace=self._namespace)
+            client.create(service)
```

Comment:
Is there a reason to delete this at all?  iiuc `_delete_and_create_service` does the delete+create to allow for renaming (it `client.get()`s the service called `self._app`, then creates a service called `self.service_name`).  Here, we get, "change" the name (but replacing it with the same name we `client.get()`), then create.  This is effectively a noop

Reply:
you're right, I may have duplicated the logic from `_delete_and_create_service` assuming my goal is the same which apparently isn't, it should be updated now not to delete the lb and only create it if it's not there

regarding your other comment, even tho it's not instant in all my tests (on local and on CI) the ip was always generated way before the pebble ready event, I think in cases of a public provider it might take a while to provision and will indeed introduce a period of unavailability @sed-i @simskij wdyt?

Reply:
Yes, it should not delete the lb if it exists as that would defeat the point for making the change in the first place. 

> I think too that externalip generation for a loadbalancer is not instant, but can instead take some time for something to be provisioned. So this might also be introducing periods where the service is unavailable

The juju-created service will still be there, but yes - you're right. The service will be unavailable for as long as it takes for traefik to get an external IP. That is not avoidable.